README.rst

```diff
@@ -453,207 +517,252 @@ Navigating the Containers
+  remarks about the customizing the sort order.)
+  The return
   value is a pair whose first entry is boolean and indicates whether
   the insertion was new (i.e., the key was not previously present) and
-  the second entry is the token of the new entry.
+  the second entry is the semitoken of the new entry. 
+  Time: O(*c* log *n*)
+
+``push!(m,k)``
+  Argument ``m`` is a SortedSet and ``k`` is a key.
+  This inserts the key into
+  the container.  If the key is already present in a
+  this overwrites
+  the old value.  (This is not necessarily a no-op; see below for 
+  remarks about the customizing the sort order.)
+  The return
+  value is ``m``.
   Time: O(*c* log *n*)
 
 
-``delete!(i)``
-  Argument ``i`` is a token.
-  This operation deletes the item addressed by ``i``.
+``delete!((m,s))``
+  Argument ``(m,s)`` is a token.
+  This operation deletes the item addressed by ``(m,s)``.
   It is an error to call
   this on an entry that has already been deleted or on the
   before-start or past-end tokens.  After this operation is 
-  complete, ``i`` is an invalid token and cannot be used in
+  complete, ``(m,s)`` is an invalid token and cannot be used in
   any further operations.
   Time: O(log *n*)
 
 ``delete!(m,k)``
-  Argument ``m`` is a SortedDict and
+  Argument ``m`` is a SortedDict or SortedSet and
   ``k`` is a key.  This operation deletes the item
   whose key is ``k``.  It is a  ``KeyError``
   if ``k`` is not a key of an item in the container.
   After this operation is 
   complete, any token addressing the deleted item is invalid.
+  Returns ``m``.
   Time: O(*c* log *n*)
 
+
 ``pop!(m,k)``
-  Deletes the item with key ``k`` in SortedDict ``m`` and returns
-  the value that was associated with ``k``.  A ``KeyError`` results
+  Deletes the item with key ``k`` in SortedDict or SortedSet ``m`` 
+  and returns
+  the value that was associated with ``k`` in the
+  case of SortedDict or ``k`` itself in the case of SortedSet.
+  A ``KeyError`` results
   if ``k`` is not in ``m``.
   Time: O(*c* log *n*)
 
-``m[st]``
-  If ``st`` is a semitoken (extracted from a token for 
-  SortedDict ``m`` via the ``semi`` function
-  below), then ``m[st]`` refers to
+``pop!(m)``
+  Deletes the item with first key in SortedSet ``m`` and
+  returns the key.  A ``BoundsError`` results if ``m`` is empty.
+  Time: O(*c* log *n*)
+
+``m[SDSemiToken, st], m[SMDSemiToken, st]``
+  If ``st`` is a semitoken and ``m`` is a SortedDict,
+  then ``m[SDSemiToken, st]`` refers to
   the value field of the (key,value) pair that the full
   token refers to.  This expression may occur on either side of an
-  assignment statement.  Time: O(1)
+  assignment statement.  The syntax ``m[SMDSemiToken, st]`` serves
+  the same function for a SortedMultiDict.  The keywords
+  ``SDSemiToken`` and ``SMDSemiToken`` should be inserted literally
+  into the operation; their purpose is to disambiguate this
+  subscripting operation from ``m[k]`` where ``k`` is a key.
```

Comment:
This shouldn't be necessary, at least technically, unless the dictionary keys themselves are `SDSemiToken`s.  I'll comment on the implementation below.